src/rules/flake8_logging_format/rules.rs

```diff
@@ -0,0 +1,238 @@
+                        LoggingPercentFormat,
+                        Range::from_located(msg),
+                    ));
+                }
+            }
+            _ => {}
+        },
+        // Check for f-strings
+        ExprKind::JoinedStr { .. } => {
+            if checker.settings.rules.enabled(&Rule::LoggingFString) {
+                checker
+                    .diagnostics
+                    .push(Diagnostic::new(LoggingFString, Range::from_located(msg)));
+            }
+        }
+        // Check for .format() calls
+        ExprKind::Call { func, .. } => {
+            if checker.settings.rules.enabled(&Rule::LoggingStringFormat) {
+                if let ExprKind::Attribute { value, attr, .. } = &func.node {
+                    if attr == "format" && matches!(value.node, ExprKind::Constant { .. }) {
+                        checker.diagnostics.push(Diagnostic::new(
+                            LoggingStringFormat,
+                            Range::from_located(msg),
+                        ));
+                    }
+                }
+            }
+        }
+        _ => {}
+    }
+}
+
+/// Check contents of the `extra` argument to logging calls
+fn check_log_record_attr_clash(checker: &mut Checker, extra: &Keyword) {
+    match &extra.node.value.node {
+        ExprKind::Dict { keys, .. } => {
+            for key in keys {
+                if let Some(key) = &key {
+                    if let ExprKind::Constant {
+                        value: Constant::Str(string),
+                        ..
+                    } = &key.node
+                    {
+                        if RESERVER_ATTRS.contains(&string.as_str()) {
+                            checker.diagnostics.push(Diagnostic::new(
+                                LoggingExtraAttrClash(string.to_string()),
+                                Range::from_located(key),
+                            ));
+                        }
+                    }
+                }
+            }
+        }
+        ExprKind::Call { func, keywords, .. } => {
+            if let ExprKind::Name { id, .. } = &func.node {
+                if id == "dict" {
+                    for keyword in keywords {
+                        if let Some(key) = &keyword.node.arg {
+                            if RESERVER_ATTRS.contains(&key.as_str()) {
+                                checker.diagnostics.push(Diagnostic::new(
+                                    LoggingExtraAttrClash(key.to_string()),
+                                    Range::from_located(keyword),
+                                ));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        _ => {}
+    }
+}
+
+/// Check logging calls for violations
+pub fn logging_call(checker: &mut Checker, func: &Expr, args: &[Expr], keywords: &[Keyword]) {
+    if let ExprKind::Attribute { value, attr, .. } = &func.node {
```

Comment:
In `src/rules/tryceratops/rules/error_instead_of_exception.rs`, I used some heuristics to detect whether the `value` is that of a logger (by matching on names, like `logging`, `logger`, `self.logger`, `foo_logger`, etc.). I'd like to be consistent between that rule and here. Any opinion on what's preferable?

Reply:
I followed the plugin here to just whitelist `parser` and `warnings` but it probably makes sense to make this more robust with some `is_logger_candidate` that incorporates your heuristic additionally. Happy to try out the refactor.

Reply:
That would be great, thanks!

Reply:
[29b19c6](https://github.com/charliermarsh/ruff/pull/2150/commits/29b19c6c554052dd63cf23c7e4ba0bf69894b729)